src/core/cgroup.c

```diff
@@ -45,10 +46,15 @@
 
 #define CGROUP_CPU_QUOTA_DEFAULT_PERIOD_USEC ((usec_t) 100 * USEC_PER_MSEC)
 
-/* Returns the log level to use when cgroup attribute writes fail. When an attribute is missing or we have access
- * problems we downgrade to LOG_DEBUG. This is supposed to be nice to container managers and kernels which want to mask
- * out specific attributes from us. */
-#define LOG_LEVEL_CGROUP_WRITE(r) (IN_SET(abs(r), ENOENT, EROFS, EACCES, EPERM) ? LOG_DEBUG : LOG_WARNING)
+/* Returns the log level to use when cgroup attribute writes fail.
+ * We downgrade to LOG_DEBUG when the attribute is missing or we have access problems (this is supposed to be
+ * nice to container managers and kernels which want to mask out specific attributes from us). Downgrade also
+ * on optional list of errors. */
+#define LOG_LEVEL_CGROUP_WRITE(r, ...) (IN_SET(abs(r), ENOENT, EROFS, EACCES, EPERM, ##__VA_ARGS__) ? LOG_DEBUG : LOG_WARNING)
+
+static const char *cgroup_jump_versions[_CGROUP_JUMP_MAX] = {
+        "5.4", /* 795fe54c2a82 ("bfq: Add per-device weight"). */
```

Comment:
We don't usually do version comparison as enterprise kernels backport all the stuff - I'd say a static bool supported = true/false would be enough? log once, then skip it forever

Reply:
FTR, there's [a precedent](https://github.com/systemd/systemd/blob/682a9f2e88dc79531c13973fee50b252447381dd/src/basic/cgroup-util.c#L2254). I added the version check (not much excited about that neither) to properly handle `EINVAL`s on the newer kernels.

OTOH (when checking the current kernel now), if we assume only the systemd well-formatted input, there should be possible only `ENODEV`, `EOPNOTSUPP`, `ERANGE` (avoided by scaling), rarely `EBUSY` or `ENOMEM`. I.e. `EINVAL` could be demoted to debug in any kernel.

I'll simplify the check then.

Reply:
Sounds good, thanks